telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java

```diff
@@ -199,7 +200,7 @@ private void scheduleBatchSend(
       scheduler.schedule(
           batch.size(), () -> sendWithErrorHandling(sender, batch, backoff), waitTime, timeUnit);
     } catch (RejectedExecutionException e) {
-      LOG.error("Problem scheduling batch : " + e.getMessage());
+      notificationHandler.noticeError("Problem scheduling batch : ", e, batch);
```

Comment:
Since the notificationHandler is not final and can be set using the with method, it probably makes sense to put in null checks.